### ISSUE TITLE
Update xed

### ISSUE THREAD

**Issue by xusheng6** [Vector35/binaryninja-api#4202]
It has been a while since we last updated the xed build, and now we are bothered by some xed bugs:

1. https://github.com/Vector35/binaryninja-api/issues/4028
2. https://github.com/Vector35/binaryninja-api/issues/4033
3. https://github.com/Vector35/binaryninja-api/issues/4617

We should check and see if these are fixed in the latest xed build. If not, we should report them upstream

**Comment by xusheng6** [comment]
I tested the latest xed and none of the three issues are fixed. I feel like we should upstream these bugs and ask xed folks to fix them so that we can benefit from it.

Also, one aspect that is making xed complicated is that we need to generate these type info for the intrinsics: https://github.com/Vector35/arch-x86/blob/master/x86_intrinsic_cached_input_types.include. I left comment in https://github.com/Vector35/arch-x86/tree/master/code_generator so I do not need to re-invent the wheels, but it is taking me some time to actually get the patch working again, on top of the latest xed. If possible, I would also love to see a way to dump these types in a more straightforward way.

**Comment by xusheng6** [comment]
Update on these:

1 is believed to be a bug in xed, and I have created an issue for it: https://github.com/intelxed/xed/issues/319
2 is the intended behavior (though slightly less than desired), and I have closed it
3 is still being worked on, and I have ruled out the possibility of a xed bug. And it is probably due to some problems in our xed build process. So updateing xed would not help this.

In other words, updating xed itself would not help any these problems right now, so I am giving this issue a lower priority

**Comment by xusheng6** [comment]
I would like to defer the xed update until https://github.com/intelxed/xed/issues/319 is fixed -- which make sense because otherwise the update would not bring about any known improvements to the x86 support. I kinda expect the xed folks will be able to fix it relatively soon, so the wait would not be too long

**Comment by xusheng6** [comment]
Xed released https://github.com/intelxed/xed/commit/6d87b5481aa53b5ab1fc2b5a5622759c46746bf9 which fixes the "push r16" bug above

**Comment by xusheng6** [comment]
I am closing this now since the only issue left within it is https://github.com/Vector35/binaryninja-api/issues/4028
